src/index.cpp

```diff
@@ -603,7 +620,7 @@ void Index<T, TagT, LabelT>::load(const char *filename, uint32_t num_threads, ui
     {
         _label_map = load_label_map(labels_map_file);
         parse_label_file(labels_file, label_num_pts);
-        assert(label_num_pts == data_file_num_pts);
+        assert(label_num_pts == data_file_num_pts - _num_frozen_pts);
```

Comment:
if we are loading an index, wont the labels file contain label info for the frozen points also?

Reply:
I believe this was added by David, I can ask him the reasoning behind this. What you said more sense.

Reply:
we need to make sure all sizes check out correctly. _pts_to_labels is set to size of total internal points in resize operation. Here we are checking for data_file_num_pts - num_frozen_points, etc. Need to make these consistent.